spray-json/shared/src/main/scala/spray/json/JsonParser.scala

```diff
@@ -165,51 +166,146 @@ class JsonParser(input: ParserInput, settings: JsonParserSettings = JsonParserSe
   private def DIGIT(): Boolean = cursorChar >= '0' && cursorChar <= '9' && advance()
 
   // http://tools.ietf.org/html/rfc4627#section-2.5
-  private def `string`(): Unit = {
-    if (cursorChar == '"') cursorChar = input.nextUtf8Char() else fail("'\"'")
-    sb.setLength(0)
-    while (`char`()) cursorChar = input.nextUtf8Char()
-    require('"')
+  private def `string`(): String = {
+    if (cursorChar != '"') fail("'\"'")
+    val res = stringFast(input.cursor)
     ws()
+    res
   }
 
-  private def `char`() =
-    // simple bloom-filter that quick-matches the most frequent case of characters that are ok to append
-    // (it doesn't match control chars, EOI, '"', '?', '\', 'b' and certain higher, non-ASCII chars)
-    if (((1L << cursorChar) & ((31 - cursorChar) >> 31) & 0x7ffffffbefffffffL) != 0L) appendSB(cursorChar)
-    else cursorChar match {
-      case '"' | EOI => false
-      case '\\' =>
-        advance(); `escaped`()
-      case c => (c >= ' ') && appendSB(c)
+  private var charBuffer = new Array[Char](settings.initialMaxStringCharacters)
+  private def stringFast(start: Int): String = {
+    /* Scan to the end of a simple string as fast as possible, no escapes, no unicode */
+    // so far unclear results if that is really helpful
+    /*@tailrec def parseOneFast(cursor: Int, charsRead: Int): String = {
+      val b = input.byteAt(cursor)
+      charBuffer(charsRead) = (b & 0xff).toChar
+      //println(s"At $cursor $charsRead read '${charBuffer(charsRead)}'")
+
+      if (b == '"') {
+        input.setCursor(cursor) // now at end '"'
+        advance()
+        new String(charBuffer, 0, charsRead)
+      } else if (((b - 32) ^ 60) <= 0) // trick from jsoniter-scala
```

Comment:
it is a check for escaping character and for characters that are not allowed to be represented as is without encoding: `b < ' ' || b = '\'`

Reply:
Yep, I know, pretty clever :) I printed out the complete mapping between inputs and outputs to see where it was going. It seems it didn't help much here in the end.